### ISSUE TITLE
Accuracy of BWN 

### ISSUE THREAD

**Issue by minhson** [tensorpack/tensorpack#1406]
Hello, i changed the code as below to see the difference between BWN with and without scaling factors:
```
def fw(x):
         if bitW == 32:
             return x
 
         if bitW == 1:   # BWN
             #FIXMEE = tf.stop_gradient(tf.reduce_mean(tf.abs(x)))
 
             @tf.custom_gradient
             def _sign(x):
                 #FIXMEreturn tf.where(tf.equal(x, 0), tf.ones_like(x), tf.sign(x / E)) * E, lambda dy: dy
                 return tf.where(tf.equal(x, 0), tf.ones_like(x), tf.sign(x)), lambda dy: dy
 
             return _sign(x)
 
         x = tf.tanh(x)
         x = x / tf.reduce_max(tf.abs(x)) * 0.5 + 0.5
         return 2 * quantize(x, bitW) - 1
```
However, the difference is only 2% in val-error-top1.
In the XNOR-Net paper, the difference is a lot as below table (taken from XNOR-Net paper):
![image](https://user-images.githubusercontent.com/8142092/76192618-9917b080-6225-11ea-8569-1d10d7096713.png)
Do you know why this happens?
Thanks you in advance!

**Comment by ppwwyyxx** [comment]
I do not know. I would attribute this to bad historical implementations.
